src/main/java/org/breedinginsight/brapps/importer/services/FileImportService.java

```diff
@@ -381,53 +381,65 @@ public ImportResponse updateUpload(UUID programId, UUID uploadId, AuthenticatedU
+            importDAO.update(upload);
+        }
 
-
-        // Spin off new process for processing the file
-        CompletableFuture.supplyAsync(() -> {
-            try {
-                importService.process(brAPIImportList, data, program, upload, user, commit);
-            } catch (UnprocessableEntityException e){
-                log.error(e.getMessage());
-                ImportProgress progress = upload.getProgress();
-                progress.setStatuscode((short) HttpStatus.UNPROCESSABLE_ENTITY.getCode());
-                progress.setMessage(e.getMessage());
-                progress.setUpdatedBy(actingUser.getId());
-                importDAO.update(upload);
-            } catch (DoesNotExistException e)  {
-                log.error(e.getMessage());
-                ImportProgress progress = upload.getProgress();
-                progress.setStatuscode((short) HttpStatus.NOT_FOUND.getCode());
-                progress.setMessage(e.getMessage());
-                progress.setUpdatedBy(actingUser.getId());
-                importDAO.update(upload);
-            } catch (HttpStatusException e) {
-                log.error(e.getMessage());
-                ImportProgress progress = upload.getProgress();
-                progress.setStatuscode((short) e.getStatus().getCode());
-                progress.setMessage(e.getMessage());
-                progress.setUpdatedBy(actingUser.getId());
-                importDAO.update(upload);
-            } catch (ValidatorException e) {
-                log.error("ValidatorException");
-                ImportProgress progress = upload.getProgress();
-                progress.setStatuscode((short) HttpStatus.UNPROCESSABLE_ENTITY.getCode());
-                progress.setMessage("Multiple Errors");
-                String json = (new JSON()).getGson().toJson(e.getErrors());
-                progress.setBody(JSONB.valueOf(json));
-                progress.setUpdatedBy(actingUser.getId());
-                importDAO.update(upload);
-            } catch (Exception e) {
-                log.error(e.getMessage());
-                ImportProgress progress = upload.getProgress();
-                progress.setStatuscode((short) HttpStatus.INTERNAL_SERVER_ERROR.getCode());
-                // TODO: Probably don't want to return this message. But do it for now
-                progress.setMessage(e.getMessage());
-                progress.setUpdatedBy(actingUser.getId());
-                importDAO.update(upload);
-            }
-            return null;
-        });
+        if (brAPIImportList != null) {
+            // Spin off new process for processing the file
+            List<BrAPIImport> finalBrAPIImportList = brAPIImportList;
+            CompletableFuture.supplyAsync(() -> {
+                try {
+                    importService.process(finalBrAPIImportList, data, program, upload, user, commit);
+                } catch (UnprocessableEntityException e){
```

Comment:
Same thing as above with the logging messages in these `catch` blocks

Reply:
Pushed changes.